### PR TITLE
Fix 4949, support closure bind with class-string

### PR DESCRIPTION
**Pull request by mvorisek**
fixes https://github.com/phpstan/phpstan/issues/4949

### CODE REVIEW

**Comment by mvorisek** [comment]
@ondrejmirtes feedback fixed, can we merge this fix?

**Comment by mvorisek** [comment]
I belive this PR is done, can you please review and merge? Thanks.

**Comment by mvorisek** [comment]
@ondrejmirtes can you please review this PR?

**Comment by ondrejmirtes** [comment]
@mvorisek You're in the same pool as everyone else so you have to wait
B4D02675-A371-4A3D-A7A1-B4B7ED867255
.

File: src/Analyser/NodeScopeResolver.php [hunk 1]
```diff
@@ -2111,17 +2112,15 @@ static function () use ($scope, $expr): MutatingScope {
...
-									&& $argValue->class instanceof Name
-								) {
-									$scopeClass = $scope->resolveName($argValue->class);
-									$thisType = new ObjectType($scopeClass);
 								} elseif ($argValueType instanceof ConstantStringType) {
 									$scopeClass = $argValueType->getValue();
 									$thisType = new ObjectType($scopeClass);
+								} elseif (
+									$argValueType instanceof GenericClassStringType
+									&& $argValueType->getGenericType() instanceof ObjectType
```
Comment (ondrejmirtes) [inline src/Analyser/NodeScopeResolver.php:2120]: instanceof TypeWithClassName would be better. Otherwise 👍 .

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: src/Analyser/NodeScopeResolver.php (hunk 1 above)
Comment (mvorisek) [reply src/Analyser/NodeScopeResolver.php]: done, 👍

**Comment by ondrejmirtes** [comment]
Thank you.

**Pull request merged**